File: admin/validation/tag.ts
import { z } from "zod";

export const TagSchema = z.object({

    name:z.string().trim().min(3),
    slug: z.string().trim().min(3).optional(),    
    description:z.string().trim().max(200),
    status:z.boolean(),
    sort_order:z.number().int().min(0),

});

File: backend/dto/tag.go
package dto

type CreateTagRequest struct {
	Name        string `json:"name" binding:"required,min=2,max=100"`
	Description string `json:"description"`
	Status      bool   `json:"status"`
	SortOrder   int    `json:"sort_order"`
}

type UpdateTagRequest struct {
	Name        string `json:"name" binding:"required,min=2,max=100"`
	Description string `json:"description"`
	Status      bool   `json:"status"`
	SortOrder   int    `json:"sort_order"`
}
